clustered/server/src/test/java/org/ehcache/clustered/server/store/ClusterTierPassiveEntityTest.java

```diff
@@ -166,6 +169,39 @@ public void testPassiveTracksMessageDuplication() throws Exception {
     assertThat(passiveEntity.getStateService().getStore(passiveEntity.getStoreIdentifier()).get(2).isEmpty(), is(false));
   }
 
+  @Test
+  public void testOversizeReplaceAtHeadMessage() throws Exception {
+    ClusterTierPassiveEntity passiveEntity = new ClusterTierPassiveEntity(defaultRegistry, defaultConfiguration, DEFAULT_MAPPER);
+    passiveEntity.createNew();
+    TestInvokeContext context = new TestInvokeContext();
+
+    int key = 2;
+
+    Chain chain = Util.getChain(true, createPayload(1L));
+    PassiveReplicationMessage message = new PassiveReplicationMessage.ChainReplicationMessage(key, chain, 2L, 1L, 3L);
+    passiveEntity.invokePassive(context, message);
+
+    Chain oversizeChain = Util.getChain(true, createPayload(2L, 1024 * 1024));
+    ServerStoreOpMessage.ReplaceAtHeadMessage oversizeMsg = new ServerStoreOpMessage.ReplaceAtHeadMessage(key, chain, oversizeChain);
+    passiveEntity.invokePassive(context, oversizeMsg);
+    // Should be evicted, the value is oversize.
+    assertThat(passiveEntity.getStateService().getStore(passiveEntity.getStoreIdentifier()).get(key).isEmpty(), is(true));
+  }
+
+  @Test
+  public void testOversizeChainReplicationMessage() throws Exception {
+    ClusterTierPassiveEntity passiveEntity = new ClusterTierPassiveEntity(defaultRegistry, defaultConfiguration, DEFAULT_MAPPER);
+    passiveEntity.createNew();
+    TestInvokeContext context = new TestInvokeContext();
+
```

Comment:
Unfortunately can't put a small value first, as it results in a successful eviction afterwards so no OSME is thrown.